src/components/branching.tsx

```diff
@@ -73,8 +73,8 @@ export const BranchingComponent: React.FC = () => {
 			<ReactFlow
 				nodes={nodes}
 				edges={edges}
-				onNodesChange={setNodes}
-				onEdgesChange={setEdges}
+				onNodesChange={() => setNodes}
```

Comment:
setNodes and setEdges are functions that need to be passed directly as handlers.
Using the arrow function syntax like this () => setNodes is incorrect because it does not actually call setNodes with the updated nodes, but rather returns the function itself.

Reply:
Can you kindly recheck if the code in current state (with `onNodesChange={setNodes}`) works fine? Actually it's throwing TSC error on my end and also if I bypass that using `any` it fails due to ReactFlow internal error during runtime in the browser. 

So I have updated it in my next commit https://github.com/massihx/chat-branching/pull/1/commits/b5363d1bf96560eb60e0534400bb4d2376b47882 as per [the suggested way mentioned in the docs](https://reactflow.dev/api-reference/hooks/use-nodes-state).